devices/src/virtio/block.rs

```diff
@@ -393,7 +393,11 @@ impl BlockEpollHandler {
 }
 
 impl EpollHandler for BlockEpollHandler {
-    fn handle_event(&mut self, device_event: DeviceEventT) -> result::Result<(), DeviceError> {
+    fn handle_event(
+        &mut self,
+        device_event: DeviceEventT,
+        _evset: epoll::Events,
```

Comment:
Why is `_evset` needed in the device handler?

Reply:
It's not needed in this particular handler (for the block device). It might, however, be needed in others (such as the vsock handler), since it carries the event type (i.e. EPOLLIN, EPOLLOUT, EPOLLHUP, etc).

Reply:
I suspected as much. In that case I suggest that it be brought in when and if (you said might) needed.

Reply:
I said "might," because the vsock epoll handler could do without, but that would break the generality of the vsock backend trait.

Also, it's generally good practice to not hide away functionality, when developing upper level wrappers. In this case, our devices are free to register an FD and a set of events under the main epoll loop, but only get notified about the triggered FD (via that token thingy), not about the event itself. So, if a device were to register an FD for both EPOLLIN and EPOLLOUT, it wouldn't have any way of figuring out which of the events had been triggered prior to entering `handle_event()`.

Reply:
Usually upper layers do hide unneeded functionality and offer an abstract layer with only needed functionality like `fork()` does for `clone()` and many other examples.

I our particular case, if this is definitely going to be used in vsock, then I'm fine with it going in now as well as when actually used (since vsock is just around the corner anyway).

Reply:
> Usually upper layers do hide unneeded functionality and offer an abstract layer with only needed functionality like fork() does for clone() and many other examples.

Right, I went a bit overboard with that generic statement :). The point I was trying to make was that, with our model, the epoll dispatch (Vmm::EpollContext), allows the devices to shoot themselves in the foot, since:
- it lets the device register for various events (the device is the one calling `epoll_ctl` and registering the individual epoll events - EPOLLIN, EPOLLOUT, etc)
- when delivering the event notification (by calling `handle_event()`), the vmm withholds the actual event, so the device has no way of knowing if it's supposed to handle, for instance, a read or a write event.

Reply:
Ok, I see what you mean. In that case I agree, if the device can register multiple types of events, then it needs to get the type as well in its callback/handler.